### ISSUE TITLE
Determine Social Provider Login Support (Facebook, Google) with StaticWebApps & App Service

### ISSUE THREAD

**Issue by seantleonard** [Azure/data-api-builder#719]
Evaluate whether social providers include roles in their access tokens, i.e. via a `role` claim.
- If role claim(s) are included:
  - Can an end user can arbitrarily add themselves to roles or is that capability limited to FB app/enterprise admins?
    - If users can arbitrarily add themselves to roles, ensure we do NOT honor those roles to determine access in DAB engine and see whether SWA/AppService passes those roles through in the EasyAuth payload.
    - If just enterprise admins can manage roles/role assignments, no issues.
      - Check whether SWA/AppService passes those roles through in the EasyAuth payload. If not, developers must manage roles through SWA Azure Function Integration.
- No Role claims included:
  - Developers must manage roles through SWA Azure Functions (preview) Integration.

**Comment by seantleonard** [comment]
I would suggest we defer support for social identities (FB, Google, etc), to identity providers like Azure AD, Okta, and Auth0. Those identity providers provide support to connect with social providers to facilitate account linking, consolidation, and token augmentation such that an access token with consistently provided claims can be provided to DAB.

**Comment by aaronpowell** [comment]
I've got an example of how to setup [SWA + Auth0](https://www.aaron-powell.com/posts/2021-05-13-using-auth0-with-static-web-apps/) and [SWA + Okta](https://www.aaron-powell.com/posts/2021-05-13-using-okta-with-static-web-apps/) if they would be of use.

When it boils down to it though, the SWA backends don't know what the identity provider is, they only see the token provided.

**Comment by seantleonard** [comment]
Yeah those would be helpful. I envision social provider support to be handled by either SWA, AppService, or azureAd/okta/auth0 directly. 

Essentially something like azure ad b2c lets users sign in and register accounts in a b2c tenant with their social provider. Behind the scenes b2c creates and manages its own user record and the user record has attributes which associate the social media account’s guid to the b2c account.

ultimately, DAB would not have any config where social provider metadata is set

**Comment by aaronpowell** [comment]
SWA is using the same engine as AppService, EasyAuth, so that might make it easier for testing as you could host DAB in AppService, enable EasyAuth and inspect what comes in as the payload*.

*In theory - I've never tried that, I just know that that is all the plumbing.
